packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/dataStreamSearch/workspaceSelector/workspaceSelector.tsx

```diff
@@ -16,16 +18,34 @@ export interface WorkspaceSelectorProps {
   control: Control<SearchFields>;
   client: IoTTwinMakerClient;
   OnGettingError: (isError: boolean) => void;
+  setValue: UseFormSetValue<SearchFields>;
 }
 
-export const WorkspaceSelector = ({ control, client, OnGettingError }: WorkspaceSelectorProps) => {
+export const WorkspaceSelector = ({ control, client, OnGettingError, setValue }: WorkspaceSelectorProps) => {
   const { workspaces, status } = useWorkspaces({ client });
   const workspaceOptions = createWorkspaceOptions(workspaces);
 
+  const [selectedWorkspace, setSelectedWorkspace] = useLocalStorage<OptionDefinition | null>('selectedWorkspace', null);
+  if (selectedWorkspace) setValue('workspace', selectedWorkspace);
```

Comment:
Do you need to set the value of workspace on every render ?

Reply:
Yes, without this we cannot avoid the flickering issue between renders.